File: Trabalho Prático/TrabalhoPG/ContadorParesImpares.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Quantos números deseja inserir? ");
        int quantidade = int.Parse(Console.ReadLine());

        ContarParesImpares(quantidade);
    }

    static void ContarParesImpares(int quantidade)
    {
        int pares = 0, impares = 0;

        for (int i = 0; i < quantidade; i++)
        {
            Console.Write($"Digite o {i + 1}º número: ");
            int numero = int.Parse(Console.ReadLine());

            if (numero % 2 == 0)
                pares++;
            else
                impares++;
        }

        ExibirResultado(pares, impares);
    }

    static void ExibirResultado(int pares, int impares)
    {
        Console.WriteLine("\n===== Resultado =====");
        Console.WriteLine($"Quantidade de números pares: {pares}");
        Console.WriteLine($"Quantidade de números ímpares: {impares}");
    }
}

File: Trabalho Prático/TrabalhoPG/contador_pares_impares.c
#include <stdio.h>

void contarParesImpares(int quantidade);
void exibirResultado(int pares, int impares);

int main()
{
    int quantidade;

    printf("Quantos números deseja inserir? ");
    scanf("%d", &quantidade);

    contarParesImpares(quantidade);

    return 0;
}

void contarParesImpares(int quantidade)
{
    int pares = 0, impares = 0, numero;

    for (int i = 0; i < quantidade; i++)
    {
        printf("Digite o %dº número: ", i + 1);
        scanf("%d", &numero);

        if (numero % 2 == 0)
        {
            pares++;
        }
        else
        {
            impares++;
        }
    }

    exibirResultado(pares, impares);
}

void exibirResultado(int pares, int impares)
{
    printf("\n===== Resultado =====\n");
    printf("Quantidade de números pares: %d\n", pares);
    printf("Quantidade de números ímpares: %d\n", impares);
}
